libraries/AP_Soaring/SoaringController.cpp

```diff
@@ -0,0 +1,407 @@
+#include "SoaringController.h"
+
+extern const AP_HAL::HAL& hal;
+
+
+// ArduSoar parameters
+const AP_Param::GroupInfo SoaringController::var_info[] = {
+    // @Param: ACTIVE
+    // @DisplayName: Is the soaring mode active or not
+    // @Description: Toggles the soaring mode on and off
+    // @Units: boolean
+    // @Range: 0 1
+    // @User: Advanced
+    AP_GROUPINFO("ACTIVE", 1, SoaringController, soar_active, 0),
```

Comment:
If thats the case we should definitely be using an ENABLE parameter. The advantage to using an ENABLE parameter for this is that when you haven't enabled soaring the rest of the soaring parameters wouldn't be sent to the GCS/shown to the user. It helps limit what is shown to the user, and improves connection times. You can see an example on the terrain library here: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Terrain/AP_Terrain.cpp#L43